nestedquoteremover-me/src/compose.js

```diff
@@ -77,14 +77,29 @@ function processHtmlBlockquotes(nodes, replyHeaderHandling, maxDepth, depth) {
     for (let node of nodes) {
         if (node.nodeType === Node.ELEMENT_NODE &&
             node.nodeName === "BLOCKQUOTE" &&
-            node.hasAttribute("type") && node.getAttribute("type") === "cite") {
+            (node.hasAttribute("type") && node.getAttribute("type") === "cite" || // regular quote
+            node.hasAttribute("class") && node.getAttribute("class") === "gmail_quote")) // gmail web interface quote
+        {
             if (depth > maxDepth) {
                 node.remove();
             } else {
                 processHtmlBlockquotes(node.childNodes, replyHeaderHandling, maxDepth, depth + 1);
             }
         }
 
+        else if (node.nodeType === Node.ELEMENT_NODE &&
+            node.nodeName === "DIV" &&
```

Comment:
Yes, it is possible when you already have a long conversation with multiple quotations from different clients (gmail web/regular different email clients).
This patch actually adds gmail web quote handling and fixes support of other clients' quotes.

Reply:
I've simplified the whole section and pushed it to the dev-branch.
Please let me know if this still works for you.
Thank you.

Reply:
The new version doesn't work for this exact email with maximum quote depth allowed > 1.
Try it yourself.

Reply:
Commit a16a2197113af97e3e6c6bf8222227bdc9b614fa doesn't work correctly on this email (with maximum quote depth allowed = 3 it strips down to 1 quotation), and ce5808b659bd69603e35c2580f13275883174378 does not work at all with quote depth allowed > 1.

Reply:
I just started Thunderbird...
File -> Open -> Saved Message -> [selected your EML-file ] -> Reply

All previous (nested) quotes got deleted. As expected.
Only the last message remained in quoted form. As expected.

Fun fact... all of this worked even without the new changes from this dev-branch.

(Importing the .eml-File maybe changed the HTML???)

Any way... I can't reproduce your problem. :cry:

Reply:
NestedQuote Remover → Preferences → Maximum quote depth allowed: 3
File -> Open -> Saved Message -> [EML-file] -> Reply. NestedQuote remover button = nothing happens.

Reply:
Sorry... I really can't see ANY issue here.
I've tried to reproduce your problem.

I've set the "max allowed quote depth" to 3:
![screenshot1](https://user-images.githubusercontent.com/3408832/133923491-84f23759-e7b1-47cc-861c-c8d1571f082e.png)

Then opened your sample email, which contains a lot of nested quotes:
![screenshot2](https://user-images.githubusercontent.com/3408832/133923535-66ac09a3-a9b7-432a-ba1a-c4a1efa05d8c.png)

After pressing the reply-button:
![screenshot3](https://user-images.githubusercontent.com/3408832/133923552-4066bc7a-8661-4eb8-a85d-787eba26f5b7.png)

After pressing the "NestedQuoteRemover" button:
![screenshot4](https://user-images.githubusercontent.com/3408832/133923566-79819821-c940-4a3d-b4f8-8f5803cb762d.png)

Works exactly as it should.
Everything beyond level 3 quotes got removed.
All without any of your code changes.

Reply:
Well, I don't know what to say. Are you _completely_ sure you use the stock unaltered version?
I've just removed the extension, downloaded it again from Mozilla Store, set depth=3, and it does nothing on this email.
Try again please. Make sure you use extension from the store.

Reply:
Yes, I'm **absolutely** sure that I've used the current release (1.3.0) of _NestedQuote Remover_, which does not contain any of your modifications.

Reply:
That's very strange, do you have some kind of Thunderbird configuration/tag stripper?
This is how this email looks for me, notice the color difference between yours and mine:

![image](https://user-images.githubusercontent.com/3054729/133933176-1627ed1b-0fc1-4bab-baea-d8fdf399d637.png)

Reply:
The white stripes are `div`s I believe.

Reply:
Just to make sure, I've tested in Thunderbird 91.1.0 on a fresh profile without any configuration.
The extension does not strip anything in this email. The color strips are the same as on the screenshot above.

Reply:
Just to make sure, md5
`4a08d92ce7055dd881007703bdfe5097  Re- обход блокировки.eml`

Reply:
> Just to make sure, md5
> `4a08d92ce7055dd881007703bdfe5097 Re- обход блокировки.eml`

Yes. Same md5sum here.
What OS are you on? Windows? Linux?

Reply:
Linux. Let me try on Windows, 10 min.

Reply:
Windows 10, Thunderbird 91.1.1, just installed & totally fresh.
The extension does not remove quotes in this message. The quote color is similar to my screenshot, not yours.

Reply:
I'm on Linux too.

Reply:
I think I know the issue!
If you open the message in plain text view (view→message body as→plain text), then when you press reply, you get proper quotation, because you don't use HTML at all.
If you open the message in HTML (view→message body as→original html), then you'll have a bug.

Reply:
Well... that is kinda obvious.
I tried it in HTML-mode (see my screenshots above).
Otherwise you wouldn't even get the view with the colored bars on the left side.

Reply:
I've emailed you the video. Please watch it.

Reply:
@ValdikSS I'm sorry... but I hope you understand that I'm going to let this issue rest for a while.
I really tried to reproduce the issue... but couldn't.
Your modifications to the code would introduce complexity and also may lead to unwanted side-effects (since at some point you treat simple DIVs as quotes, etc.).
Hope you understand that I'm a bit hesitant here.

Reply:
> I really tried to reproduce the issue... but couldn't.

That's because you're using "reply" button while in a plain text view. It does add color bars, just like on your screenshot (no non-colored bars). In this mode the extension works correctly. However, it doesn't while replying in HTML mode.

Reply:
> I've emailed you the video. Please watch it.

Oh. Okay... just watched it.
That was helpful. :smile:
I'll get back to it. (Probably next weekend.)
Thanks!

Reply:
Hello, any updates?

Reply:
No... sorry.
I've opened another [issue](https://github.com/4ch1m/mozext/issues/50) to work on this.
Haven't found the time for it so far. :disappointed: